### PR TITLE
Don't use D-APT for compiler downloads

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**
It doesn't store old compiler versions

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
FYI @mathias-lang-sociomantic

**Comment by mathias-lang-sociomantic** [comment]
Travis says no

**Comment by mathias-lang-sociomantic** [comment]
Wait, can't you get the D compiler from Travis ?

**Comment by mathias-lang-sociomantic** [comment]
https://github.com/rejectedsoftware/vibe.d/blob/1a082c5104229171a6d2bdbd2e1c610b167e2b82/.travis.yml#L8-L17

**Comment by mihails-strasuns-sociomantic** [comment]
AFAIR it doesn't work with the docker image as an executor, I asked @leandro-lucarella-sociomantic about it before but forgot the answer :)

**Comment by mihails-strasuns-sociomantic** [comment]
Should be green now

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,3 @@
-language: d
-
```
Comment (mihails-strasuns-sociomantic) [inline .travis.yml:2]: NB: @leandro-lucarella-sociomantic this resulted in default D compiler version being installed in host (== useless action)

**Comment by mathias-lang-sociomantic** [comment]
> AFAIR it doesn't work with the docker image as an executor

Ah I forgot this part. Yes you're right.

Will merge as soon as CI finishes.

File: .travis.yml (hunk 1 above)
Comment (leandro-lucarella-sociomantic) [reply .travis.yml:2]: We should probably move this to the same scheme we'll use in ocean / tango.

Comment (mihails-strasuns-sociomantic) [reply .travis.yml:2]: Sure, as soon as that scheme is ready :)

Comment (leandro-lucarella-sociomantic) [reply .travis.yml:2]: Oh, we already are!

File: docker/dmd.sh [hunk 2]
```diff
@@ -4,17 +4,9 @@
...
-        wget http://master.dl.sourceforge.net/project/d-apt/files/d-apt.list \
-            -O /etc/apt/sources.list.d/d-apt.list
+apt -y install xdg-utils libcurl3
 
-# Make sure our packages list is updated
-apt-get update
-
-# Install dmd-bin
-apt-get install -y dmd-bin$dmd_version
+wget -O dmd.deb http://downloads.dlang.org/releases/2.x/${dmd_version}/dmd_${dmd_version}-0_amd64.deb
```
Comment (leandro-lucarella-sociomantic) [inline docker/dmd.sh:11]: No, I don't get this, why was this necessary?

Comment (mihails-strasuns-sociomantic) [reply docker/dmd.sh:11]: Downloading deb? Because d-apt doesn't keep old DMD versions, you can only install latest one at any single point of time.

Comment (leandro-lucarella-sociomantic) [reply docker/dmd.sh:11]: "It doesn't store old compiler versions", oh, OK, this is why. This is shitty, I guess we can ask jordi about it?

Comment (mihails-strasuns-sociomantic) [reply docker/dmd.sh:11]: Please try :)

Comment (leandro-lucarella-sociomantic) [reply docker/dmd.sh:11]: Wait, the versions are there:
```
$ apt-cache policy dmd-bin
dmd-bin:
  Installed: 2.072.1-0
  Candidate: 2.072.1-0
  Version table:
 *** 2.072.1-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
        100 /var/lib/dpkg/status
     2.072.0-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.071.2-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.071.1-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.071.0-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.070.2-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.070.1-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.070.0-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.069.2-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.069.1-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.069.0-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.068.2-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.068.1-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.068.0-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.067.1-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.067.0-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.066.1-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.065.0-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
     2.064.2-0 500
        500 http://master.dl.sourceforge.net/project/d-apt d-apt/main amd64 Packages
```

Comment (mihails-strasuns-sociomantic) [reply docker/dmd.sh:11]: Does installing specific version work on your local system? I had similar errors to ones in travis when tried myself.

Comment (leandro-lucarella-sociomantic) [reply docker/dmd.sh:11]: It does work, perfectly. Maybe the version number was missing the trailing, debian-specific, `-0`? This is even working in the tangort repo now: https://github.com/sociomantic-tsunami/tangort/blob/ffcb78619bb4ab6714e44d3a74770307effb1e49/docker/Dockerfile#L4

Comment (mihails-strasuns-sociomantic) [reply docker/dmd.sh:11]: Hm, you can try setting up a PR that reverts this bit and check if it gets green :)

Comment (leandro-lucarella-sociomantic) [reply docker/dmd.sh:11]: Let's see: https://github.com/sociomantic-tsunami/d1to2fix/pull/15

Comment (leandro-lucarella-sociomantic) [reply docker/dmd.sh:11]: :green_heart:
